Competition2017/src/competition/subsystems/drive/commands/DriveToPointUsingHeuristicsCommand.java

```diff
@@ -0,0 +1,120 @@
+package competition.subsystems.drive.commands;
+
+import com.google.inject.Inject;
+
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.FieldPose;
+import xbot.common.math.PIDFactory;
+import xbot.common.math.PIDManager;
+import xbot.common.math.XYPair;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class DriveToPointUsingHeuristicsCommand extends BaseDriveCommand {
+
+    private final PoseSubsystem poseSubsystem;
+    private FieldPose goalPoint;
+    
+    private final PIDManager alignToFinalHeadingPid;
+    private final PIDManager headTowardsLinePid;
+    private final PIDManager destinationPid;
+    
+    private final DoubleProperty finalHeadingEffectDistance;
+    private final DoubleProperty driveToLineEffectDistance;
+    
+    @Inject
+    public DriveToPointUsingHeuristicsCommand(
+            DriveSubsystem driveSubsystem,
+            PoseSubsystem poseSubsystem,
+            PIDFactory pidFactory,
+            XPropertyManager propMan) {
+        super(driveSubsystem);
+        this.poseSubsystem = poseSubsystem;
+        
+        alignToFinalHeadingPid = pidFactory.createPIDManager("AlignToFinalHeading", .1, 0, 0, 0, 1, -1, 2, 2, 1);
```

Comment:
consider maybe 'targetFinalHeading' instead, I think we use that more often to describe the setpoint of such a system?